tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -4947,6 +4954,51 @@ void CGMSHLSLRuntime::EmitHLSLMatrixStore(CGBuilderTy &Builder, Value *Val,
                                  Val->getType(), {DestPtr, Val}, TheModule);
 }
 
+bool CGMSHLSLRuntime::NeedHLSLMartrixCastForStoreOp(const clang::Decl* TD,
+  llvm::SmallVector<llvm::Value*, 16>& IRCallArgs) {
+
+  if (!TD || !TD->hasAttr<HLSLIntrinsicAttr>())
```

Comment:
You can use:
```
unsigned opcode = 0;
StringRef group;
if (!hlsl::GetIntrinsicOp(TD, opcode, group))
  return false;
if (opcode != (unsigned)hlsl::IntrinsicOp::MOP_Store)
  return false;
```
Instead of manually referencing the mechanism used.